library/ssl_tls.c

```diff
@@ -5991,56 +6037,165 @@ static int ssl_session_save( const mbedtls_ssl_session *session,
+            *p++ = (unsigned char)( ( start       ) & 0xFF );
+        }
+#endif /* MBEDTLS_HAVE_TIME */
+
+        /*
+         * Basic mandatory fields
+         */
+        used += 2   /* ciphersuite */
+              + 4   /* ticket_lifetime */
+              + 4   /* ticket_age_add */
+              + 1   /* key_len */
+              + 1;  /* flags */
+
+        if( used <= buf_len )
+        {
+            *p++ = (unsigned char)( ( session->ciphersuite >> 8 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ciphersuite      ) & 0xFF );
+
+            *p++ = (unsigned char)( ( session->ticket_lifetime >> 24 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_lifetime >> 16 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_lifetime >>  8 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_lifetime       ) & 0xFF );
+
+
+            *p++ = (unsigned char)( ( session->ticket_age_add >> 24 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_age_add >> 16 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_age_add >>  8 ) & 0xFF );
+            *p++ = (unsigned char)( ( session->ticket_age_add       ) & 0xFF );
+
+            *p++ = (unsigned char)( ( session->resumption_key_len      ) & 0xFF );
+
+            *p++ = (unsigned char)( ( session->ticket_flags      ) & 0xFF );
+        }
+
+        used += session->resumption_key_len;
+
+        if( used <= buf_len )
+        {
+            memcpy( p, session->key, session->resumption_key_len );
+            p += session->resumption_key_len;
+        }
+
+#if defined(MBEDTLS_SSL_CLI_C)
+        if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+        {
+            /* Store ticket itself */
+            used += 2                     /* ticket length */
+                 + session->ticket_len;   /* ticket */
+
+            if( used <= buf_len )
+            {
+                *p++ = (unsigned char)( ( session->ticket_len >> 8 ) & 0xFF );
+                *p++ = (unsigned char)( ( session->ticket_len      ) & 0xFF );
+
+                memcpy( p, session->ticket, session->ticket_len );
+                p +=  session->ticket_len;
+            }
+
+#if defined(MBEDTLS_HAVE_TIME)
+            used += 8;
+
+            if( used <= buf_len )
+            {
+                start = (uint64_t) session->ticket_received;
+
+                *p++ = (unsigned char)( ( start >> 56 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >> 48 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >> 40 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >> 32 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >> 24 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >> 16 ) & 0xFF );
+                *p++ = (unsigned char)( ( start >>  8 ) & 0xFF );
+                *p++ = (unsigned char)( ( start       ) & 0xFF );
+            }
+#endif /* MBEDTLS_HAVE_TIME */
+        }
 
```

Comment:
If we drop the certificate from the ticket, we can return here, which removes the need for some non-1.3 version checks below.

Reply:
I removed the certificate storage